examples/checkout.c

```diff
@@ -0,0 +1,229 @@
+/*
+ * libgit2 "checkout" example - shows how to perform checkouts
+ *
+ * Written by the libgit2 contributors
+ *
+ * To the extent possible under law, the author(s) have dedicated all copyright
+ * and related and neighboring rights to this software to the public domain
+ * worldwide. This software is distributed without any warranty.
+ *
+ * You should have received a copy of the CC0 Public Domain Dedication along
+ * with this software. If not, see
+ * <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+
+#include "common.h"
+#include <assert.h>
+
+/* Define the printf format specifer to use for size_t output */
+#if defined(_MSC_VER) || defined(__MINGW32__)
+#	define PRIuZ "Iu"
+#	define PRIxZ "Ix"
+#	define PRIdZ "Id"
+#else
+#	define PRIuZ "zu"
+#	define PRIxZ "zx"
+#	define PRIdZ "zd"
+#endif
+
+/**
+ * The following example demonstrates how to do checkouts with libgit2.
+ *
+ * Recognized options are :
+ *  --force: force the checkout to happen.
+ *  --[no-]progress: show checkout progress, on by default.
+ *  --perf: show performance data.
+ */
+
+typedef struct {
+	int force : 1;
+	int progress : 1;
+	int perf : 1;
+} checkout_options;
+
+static void print_usage(void)
+{
+	fprintf(stderr, "usage: checkout [options] <branch>\n"
+		"Options are :\n"
+		"  --git-dir: use the following git repository.\n"
+		"  --force: force the checkout.\n"
+		"  --[no-]progress: show checkout progress.\n"
+		"  --perf: show performance data.\n");
+	exit(1);
+}
+
+static void parse_options(const char **repo_path, checkout_options *opts, struct args_info *args)
+{
+	if (args->argc <= 1)
+		print_usage();
+
+	/* Default values */
+	opts->progress = 1;
+
+	for (args->pos = 1; args->pos < args->argc; ++args->pos) {
+		const char *curr = args->argv[args->pos];
+		int bool_arg;
+
+		if (strcmp(curr, "--") == 0) {
```

Comment:
You should probably use `!strcmp`, for consistency with the next conditional. Might confuse readers otherwise